GVFS/GVFS.Common/NamedPipes/NamedPipeServer.cs

```diff
@@ -200,10 +201,11 @@ public string ReadRequest()
             {
                 try
                 {
-                    return this.reader.ReadLine();
+                    return this.reader.ReadMessage();
                 }
-                catch (IOException)
+                catch (IOException ex)
                 {
+                    this.tracer.RelatedError($"Error reading message from NamedPipe: {ex.Message}");

```

Comment:
Minor suggestions:
- Include the full stack trace
- Create an EventMetadata and add a value for the stack trace